tests/unit/MSFT_xDisk.tests.ps1

```diff
@@ -0,0 +1,212 @@
+        Describe "$($Global:DSCResourceName)\Test-TargetResource" {
+            context 'Test matching AllocationUnitSize' {
+                Mock Get-WmiObject -mockwith {return $global:mockedWmi} -verifiable
+                Mock Get-Disk -mockwith {return $global:mockedDisk0} -verifiable
+                Mock Get-Partition -mockwith {
+                    return $Global:mockedPartition
+                } -verifiable
+                
+                Mock Get-Volume -mockwith {
+                    return $global:mockedVolume
+                } 
+                
+                $script:result = $null
+                
+                it 'calling test should not throw' {
+                    {$script:result = Test-TargetResource -DiskNumber 0 -DriveLetter 'F' -AllocationUnitSize 4096 -verbose} | should not throw
+                }
+
+                it "result should be true" {
+                    $script:result | should be $true
+                } 
+
+                it "the correct mocks were called" {
+                    Assert-VerifiableMocks
+                    Assert-MockCalled -CommandName Get-Volume -Times 0
+                }
+            }
+            context 'Test mismatched AllocationUnitSize' {
+                Mock Get-WmiObject -mockwith {return $global:mockedWmi} -verifiable
+                Mock Get-Disk -mockwith {return $global:mockedDisk0} -verifiable
+                Mock Get-Partition -mockwith {
+                    return $Global:mockedPartition
+                } -verifiable
+                
+                Mock Get-Volume -mockwith {
+                    return $global:mockedVolume
+                } 
+                
+                $script:result = $null
+                
+                it 'calling test should not throw' {
+                    {$script:result = Test-TargetResource -DiskNumber 0 -DriveLetter 'F' -AllocationUnitSize 4097 -verbose} | should not throw
+                }
+
+                # skipped due to:  https://github.com/PowerShell/xStorage/issues/22
+                it "result should be true" -skip {
+                    $script:result | should be $false
+                } 
+
+                it "the correct mocks were called" {
+                    Assert-VerifiableMocks
+                    Assert-MockCalled -CommandName Get-Volume -Times 0
+                }
+            }
+        }
+        #endregion
+
+
+        #region Function Set-TargetResource
+        Describe "$($Global:DSCResourceName)\Set-TargetResource" {
+            context 'Online Unformated disk' {
+                Mock Get-WmiObject -mockwith {return $global:mockedWmi}
+                Mock Get-Disk -mockwith {return $global:mockedDisk0Raw} -verifiable
+                Mock Get-Partition -mockwith {
+                    return $Global:mockedPartition
+                } 
+                
+                Mock Get-Volume -mockwith {
+                    return $global:mockedVolume
+                } 
```

Comment:
Better to have unified format (all in one line or all separated into three lines). Same for above places.